### PR TITLE
#173

### PR DESCRIPTION
**Pull request by evscott**
Editable form for user information which allows the user to update just their name, or their name and password.

### CODE REVIEW

File: src/redux/actions/userActions.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'PUT',
+      body: JSON.stringify(user)
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        if (json.success === false) dispatch(updateUserInfoInDbFailure());
```
Comment (gzinck) [inline src/redux/actions/userActions.js]: Yes, this is good!!

File: src/redux/tests/accomReducer.test.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import React from 'react';
```
Comment (gzinck) [inline src/redux/tests/accomReducer.test.js]: Is there a reason this import was needed? I legit have no idea if it's required or not 😂

File: src/components/user/password_change/PasswordChange.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Form } from 'react-bootstrap';
+
+/**
+ * This component simply displays a users information.
```
Comment (morgankbender) [inline src/components/user/password_change/PasswordChange.js]: This comment should be updated

File: src/components/global/ReturnButton.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ * Generic return button which accepts an onClick handler.
+ */
+export class ReturnButton extends Component {
+  render() {
+    return (
+      <OverlayTrigger
+        key={'1'}
+        placement={'left'}
+        overlay={<Tooltip>Go back?</Tooltip>}
+      >
```
[the remark this answers is not in the document]
Comment (evscott) [reply src/components/global/ReturnButton.js]: Thank you! Tooltips are pretty cool. Though, they require a lot of boiler plate code. I think a helper service which takes in an html component and a descriptive string to attach to a tooltip would be pretty cool.

File: src/redux/actions/authActions.js [hunk 5]
```diff
@@ -14,9 +15,6 @@ export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
 export const LOGOUT_REQUEST = 'LOGOUT_REQUEST';
 export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';
 export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
-export const GET_USER_INFO_REQUEST = 'GET_USER_INFO_REQUEST';
-export const GET_USER_INFO_FAILURE = 'GET_USER_INFO_FAILURE';
-export const GET_USER_INFO_SUCCESS = 'GET_USER_INFO_SUCCESS';
```
[the remark this answers is not in the document]
Comment (evscott) [reply src/redux/actions/authActions.js]: Agreed! Hah.

File: src/redux/actions/userActions.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+    return fetch(hostUrl + '/user/getinfo', {
+      headers: {
+        'Content-Type': 'application/json',
+        'x-access-token': localStorage.getItem('token')
+      },
+      method: 'get'
+    })
+      .then((response) => response.json())
+      .then((json) => {
+        if (json.length === 0) dispatch(getUserInfoFromDbFailure());
```
[the remark this answers is not in the document]
Comment (evscott) [reply src/redux/actions/userActions.js]: You're correct, I'll refactor this.

File: src/redux/actions/userActions.js (hunk 1 above)
Comment (evscott) [reply src/redux/actions/userActions.js]: Thanks!

File: src/components/user/password_change/PasswordChange.js (hunk 3 above)
Comment (evscott) [reply src/components/user/password_change/PasswordChange.js]: Ahh true. Thanks for catching this. Was recycling another components structure here.

File: src/redux/tests/accomReducer.test.js (hunk 2 above)
Comment (evscott) [reply src/redux/tests/accomReducer.test.js]: Haha I don't think so. Some of the tests were failing in this branch, so I copy and pasted the test suite from another branch in this past and this snuck in with that. I'll remove it.

File: src/redux/actions/userActions.js [hunk 7]
```diff
@@ -83,7 +83,7 @@ export function getUserInfoFromDb() {
     })
       .then((response) => response.json())
       .then((json) => {
-        if (json.length === 0) dispatch(getUserInfoFromDbFailure());
+        if (json.success === false) dispatch(getUserInfoFromDbFailure());
```
Comment (gzinck) [inline src/redux/actions/userActions.js]: Excellent.

**Pull request merged**